Checks if getenv returns proper system environment variable results

mmattel opened the pull request:
In addition to:
#15597 (Adding a final fallback for findBinaryPath)
#15546 (SMB is broken - gone from the list of available external storages)

It checks if getenv('PATH') returns empty or a result. In case of empty, a warning is printed in the `Security and Setup` section of the admin page because php is not properly setup to return system environment variables.

It is also needed to update the documentation, will be done in an seperate PR.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/6586f3b1-c712-4d7a-8d9b-0b34032fc5db) was created.

nickvergessen commented on settings/admin.php:
```diff
@@ -118,6 +118,9 @@
 $template->assign('cronErrors', $appConfig->getValue('core', 'cronErrors'));
 
+// warn if php is not setup properly to get system variables with getenv
+$template->assign('getenvServerWorking', empty(getenv('PATH')));
```
Working is `!empty` :wink: 
Please invert the logic of the name or the value

nickvergessen commented on settings/templates/admin.php:
```diff
@@ -67,6 +67,16 @@
+?>
+	<li>
+		<?php p($l->t('php does not seem to be setup properly to query system environment variables. The test with getenv("PATH") only returns an empty response.')); ?><br>
+		<?php p($l->t('Please check the documentation and your php configuration, especially when using php-fpm.')); ?>
```
which documentation? owncloud, php, os?

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/037328ec-5163-42eb-8eab-fd7713d997e0) was created.

nickvergessen commented on settings/templates/admin.php:
```diff
@@ -67,6 +67,16 @@
+?>
+	<li>
+		<?php p($l->t('php does not seem to be setup properly to query system environment variables. The test with getenv("PATH") only returns an empty response.')); ?><br>
+		<?php p($l->t('Please check the ownCloud documentation and the php configuration of your server, especially when using php-fpm.')); ?>
```
Please add a link to the page of the documentation which helps here ;)

mmattel replied:
There is currently no documentation present about that particualr topic I can reference to.

There is only one note at "Configuration notes to php.ini files" in  https://github.com/owncloud/documentation/blob/master/admin_manual/installation/source_installation.rst but this is not sufficient.

I don´t know right now if I will do this there or it may fit somewhere else.

This is what I wrote above. Will be done in an seperate PR.
(But this will be more going to the weekend)

nickvergessen replied:
@DeepDiver1975 is it okay to name "ownCloud" in respect to white-labeling in the admin section while talking about docs?
I think it is correct as we talk about the software product and not the instance name.

DeepDiver1975 replied:
simple removing the term 'ownCloud' is enough - thx

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/aaa09bbf-3653-40ba-ba0c-044ff8f7746c) was created.

mmattel commented:
I have updated the text a bit and added a printout which path is used for the php.ini files. This helps admins solving the issues easier.

Next I will do a PR on documentation and reference it to here

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/aaa09bbf-3653-40ba-ba0c-044ff8f7746c) completed: **4334 Issues, 130 Patches**

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/85e634e7-3780-4c42-a098-e1eea98cc80b) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/d932631f-3429-450b-82b0-870a496a049c) was created.

mmattel commented:
Removed the term ownCloud from the text as suggested by @DeepDiver1975 and squashed the commits.

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/85e634e7-3780-4c42-a098-e1eea98cc80b) completed: **3 new issues**

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/d932631f-3429-450b-82b0-870a496a049c) completed: **3 new issues**

mmattel commented:
@LukasReschke @oparoz @DeepDiver1975 
Any comments from you?

Note: the reason for this PR is to highlight the problem that an admin should solve even we have fallback parameters defined - but they may not meet the needs. And if this is not stated, nobody ever will recognize it.

oparoz commented:
Regarding the path where the configuration is at as it's more than likely that you will get it wrong as there is a chain of config files.
```
php.ini
php.conf.d/custom1.ini
php.conf.d/custom2.ini
global/php-fpm.ini
local/php-fpm-ini
```

So I would just remove that line

Xenopathic commented:
@oparoz Agreed - a simple pointer to the documentation will suffice.

mmattel commented:
OK, I will change that as suggested.

mmattel committed:
Checks if getenv returns proper system variable results

Updated texts and changed the variable name to match the query

Updated text + info which path for php config is used

removed the term ownCloud and squashed the commits

Updated text

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/d53f1c0c-8f2c-4aa0-8f36-16675261472f) was created.

mmattel commented:
I have also updated the corresponding PR at documentation

scrutinizer-notifier commented:
The [inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/d53f1c0c-8f2c-4aa0-8f36-16675261472f) completed: **No new issues**

Xenopathic commented:
:+1:

DeepDiver1975 commented:
:+1:

DeepDiver1975 commented:
THX
